tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/markers/MarkerSupportViewTest.java

```diff
@@ -125,6 +130,59 @@ public void limitDisabled() throws Exception {
 		assertFalse(isLimitEnabled);
 	}
 
+	@Test
+	public void markerContentGeneratorExtensionLoaded() throws Exception {
+		MarkerSupportView view = (MarkerSupportView) PlatformUI.getWorkbench().getActiveWorkbenchWindow()
+				.getActivePage().showView(PROBLEM_VIEW_ID);
+
+		MarkerContentGenerator generator = getMarkerContentGenerator(view);
+		ContentGeneratorDescriptor descriptor = MarkerSupportRegistry.getInstance()
+				.getContentGenDescriptor(generator.getId());
+		assertNotNull(descriptor);
+
+		MarkerField[] allFields = descriptor.getAllFields();
+		boolean foundAdditionalField1 = false;
+		boolean foundRecursivelyAddedField2 = false;
+		for (MarkerField field : allFields) {
+			if (ProblemKeyMarkerField.class.equals(field.getClass())) {
+				if ("Problem Key".equals(field.getName())) {
+					foundAdditionalField1 = true;
+				}
+
+				if ("Problem Key V2".equals(field.getName())) {
+					foundRecursivelyAddedField2 = true;
+				}
+			}
+		}
+		assertTrue("MarkerFields from marker content generator extensions not loaded.",
+				foundAdditionalField1);
+		assertTrue("MarkerField not loaded from recursive marker content generator extension.",
+				foundRecursivelyAddedField2);
+
+		MarkerField[] initiallyVisibleFields = descriptor.getInitialVisible();
+		boolean foundAdditionalVisbileField = false;
+		for (MarkerField field : initiallyVisibleFields) {
+			if ("Problem Key".equals(field.getName())) {
+				foundAdditionalVisbileField = true;
+			}
+		}
+		assertTrue("Visible attribute from marker fields from marker content generator extensions not loaded.",
+				foundAdditionalVisbileField);
+
+		MarkerType markerTypeFromExtension = descriptor.getType("org.eclipse.ui.tests.markers.artificial.problem");
+		assertNotNull("Marker type from marker content generator extensions not loaded.", markerTypeFromExtension);
+		assertTrue(descriptor.getMarkerTypes().contains(markerTypeFromExtension));
```

Comment:
Please add some fail message here that lists the contents of `getMarkerTypes()`.